cmd/minikube/cmd/config/open.go

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"os"
+	"text/template"
+	"time"
+
+	"github.com/docker/machine/libmachine"
+	"github.com/spf13/cobra"
+	"k8s.io/minikube/pkg/minikube/assets"
+	"k8s.io/minikube/pkg/minikube/cluster"
+	"k8s.io/minikube/pkg/minikube/constants"
+	"k8s.io/minikube/pkg/util"
+)
+
+var (
+	namespace         string
+	https             bool
+	addonsURLMode     bool
+	addonsURLFormat   string
+	addonsURLTemplate *template.Template
+)
+
+const defaultAddonsFormatTemplate = "http://{{.IP}}:{{.Port}}"
+
+var addonsOpenCmd = &cobra.Command{
+	Use:   "open ADDON_NAME",
+	Short: "Opens the addon w/ADDON_NAME within minikube (example: minikube addons open dashboard). For a list of available addons use: minikube addons list ",
+	Long:  "Opens the addon w/ADDON_NAME within minikube (example: minikube addons open dashboard). For a list of available addons use: minikube addons list ",
+	PersistentPreRun: func(cmd *cobra.Command, args []string) {
```

Comment:
You are correct, I believe it should be PreRun. I will change it in service.go cmd as well .